Source/OxyPlot/PlotModel/PlotModel.Rendering.cs

```diff
@@ -500,7 +500,8 @@ private void UpdatePlotArea(IRenderContext rc)
 
             // Find the available size for the legend box
             var availableLegendWidth = plotArea.Width;
```

Comment:
Originally, I hadn't changed the line above, but it caused items that didn't fit in the legend to be shown outside and to the right of the legend. This happened because, if the height was modified to the max, the width of the legend wasn't changed in order to accommodate all items. In other words, the legend was being rendered with the width measured for the original height, not the new height. The same thing happens with LegendMaxWidth. Items that don't fit the new legend are shown outside and to the bottom of the legend. Is this the intended behavior? What I want is to be able to set a max height and let the plot model figure out the width needed to fit all of the items. My changes do that for height but not for width because I didn't want to change any expected behavior.